### PR TITLE
Improve resource cleanup of manageiq job to use OSC since O release

### PR DESCRIPTION
**Pull request by liu-sheng**
Closes: theopenlab/openlab#101

### CODE REVIEW

**Comment by liu-sheng** [comment]
recheck

File: roles/clean-devstack-default-resources/tasks/main.yaml [hunk 1]
```diff
@@ -5,22 +5,26 @@
       set -x
       source /opt/stack/new/devstack/openrc admin admin
       export OS_DOMAIN_NAME=Default
-      if [ "{{ global_env.OS_BRANCH }}" == "master" ]; then
-          # NOTE: workaround for issue #theopenlab/openlab/issues/101
-          pip install "openstacksdk==0.17.2"
+      export OS_DOMAIN_NAME=Default
```
Comment (kiwik) [inline roles/clean-devstack-default-resources/tasks/main.yaml]: duplicate line

File: roles/clean-devstack-default-resources/tasks/main.yaml [hunk 2]
```diff
@@ -5,22 +5,26 @@
...
+          neutron router-gateway-clear router1
+          neutron router-interface-delete router1 ipv6-private-subnet
+          neutron router-interface-delete router1 private-subnet
+          neutron router-delete router1
+          openstack port delete `neutron port-list -f value -c id` || true
+      else
+          openstack router unset router1 --external-gateway
+          openstack router remove subnet router1 ipv6-private-subnet
+          openstack router remove subnet router1 private-subnet
+          openstack router delete router1
```
Comment (kiwik) [inline roles/clean-devstack-default-resources/tasks/main.yaml]: why do not delete port at here, like: line 14?

Comment (liu-sheng) [reply roles/clean-devstack-default-resources/tasks/main.yaml]: according to my testing, do not need, in testing with Mitaka, it need to delete ports

File: roles/clean-devstack-default-resources/tasks/main.yaml (hunk 1 above)
Comment (liu-sheng) [reply roles/clean-devstack-default-resources/tasks/main.yaml]: oh.. will fix

File: roles/clean-devstack-default-resources/tasks/main.yaml (hunk 2 above)
Comment (kiwik) [reply roles/clean-devstack-default-resources/tasks/main.yaml]: would we can add a comment to make it clear? that will reduce confusion.

Comment (kiwik) [reply roles/clean-devstack-default-resources/tasks/main.yaml]: And please add `Closes: theopenlab/openlab#101` into commit message

**Comment by kiwik** [comment]
Hi @liu-sheng please see my comments https://github.com/theopenlab/openlab-zuul-jobs/pull/359#discussion_r231397605 and https://github.com/theopenlab/openlab-zuul-jobs/pull/359#discussion_r231397908

File: roles/clean-devstack-default-resources/tasks/main.yaml (hunk 2 above)
Comment (liu-sheng) [reply roles/clean-devstack-default-resources/tasks/main.yaml]: ok, sure

**Pull request merged**